nodejs.js

```diff
@@ -0,0 +1,82 @@
+const express = require("express");
+const sql = require("mysql2");
+const app = express();
+const bodyParser = require('body-parser');
+const { check, validationResult } = require('express-validator');
+
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(express.static("public"));
+app.set("view engine", "ejs");
+
+//SIGN UP PAGE
+app.get("/", (req, res) => {
+  res.sendFile( __dirname + '/signuppage.html');
+});
+
+//VALIDATIONS
+app.post("/signup",[
+  check('username','Mininum length of Username is 3')
+    .isLength({ min : 3 }),
+  check('password','Mininum length of Password is 3')
+    .isLength({ min : 3 }),
+  check('email', 'Invalid Email')
+    .isEmail()
+    .normalizeEmail()
+  ],
+  (req,res) =>
+  {
+    const errors = validationResult(req);
+    if(!errors.isEmpty())
+    {
+      const alert = errors.array();
+      //console.log(alert, "errors listed above");
+      res.render('errordisplay',{ alert, alerts :'Invalid' });
+    }
+    else
+    {
+      const { email, username, password } = req.body;
+      let count = 0  ; //count counts the error
+      connection.query('INSERT INTO CustomerList(cust_name,email,password) VALUES(?, ?, ?)',[username,email,password],
+      (err)=>
+      {
+        if (err) count = count + 1;
+        const errors = validationResult(req);
+        if (count > 0)
```

Comment:
what if the query fails because of database connection problem? Will the error message say that it is duplicate email id ?